PrebidMobile/AdUnits/AdUnit.swift

```diff
@@ -157,7 +157,18 @@ public class AdUnit: NSObject, DispatcherDelegate {
         lastFetchDemandCompletion = completion
         adServerObject = adObject
         
+        
+        let timeoutHandler = DispatchWorkItem {
+            if (!self.didReceiveResponse) {
+                self.timeOutSignalSent = true
+                completion(BidInfo(resultCode: .prebidDemandTimedOut))
+                return
+            }
+        }
+        
         bidRequester.requestBids { [weak self] bidResponse, error in
+            timeoutHandler.cancel()
```

Comment:
The solution looks great! Could you please consider adding some unit tests for this?